### ISSUE TITLE
dotnet: how to represent getter/setters

### ISSUE THREAD

**Issue by williballenthin** [mandiant/capa#939]
dotnet may emit accessor/mutators (getter/setters) for some fields rather than direct field access. how do we recognize and emit these features?

**Comment by williballenthin** [comment]
need concrete example

**Comment by williballenthin** [comment]
do we need a few feature for instance property access? do we reuse `offset`?

e.g.

```yaml
- property: System.Net.FtpWebResponse.statusCode
- offset: System.Net.FtpWebResponse.statusCode
```

**Comment by mike-hunhoff** [comment]
![Screen Shot 2022-04-07 at 10 43 09 AM](https://user-images.githubusercontent.com/42192796/162254689-0717cdd5-86ad-4d6d-b20b-ab3694bbb83d.png)

**Comment by mike-hunhoff** [comment]
for imported classes we may be able to identify getter/setter by signature type; specifically if the signature is of type `Property`.
 
![Screen Shot 2022-04-07 at 10 58 02 AM](https://user-images.githubusercontent.com/42192796/162258195-00eb88c6-e29c-4251-adb2-dbeea3532163.png)

**Comment by williballenthin** [comment]
wow getters can have parameters? makes me think we might want to emit both API and property features.

**Comment by mike-hunhoff** [comment]
example:

C#:
```c#
ftpRequest.KeepAlive = false;
```

IL:
```c#
IL_0000: callvirt  instance void [System]System.Net.FtpWebRequest::set_KeepAlive(bool)
```

**Comment by mike-hunhoff** [comment]
I like the idea of adding new `property` and `field` features. it may also be useful for users to specify whether a property/field is read/written.

detection should be fairly straightforward. non-public fields are accessed via properties, which we can identify using the .NET metadata. public fields, from my understanding, are accessed directly using CIL instructions `stfld`/`stsfld`, `ldfld`/`ldflda`/`ldsfld`/`ldsflda`.

I'm unsure how to represent read/writes to properties/fields if we decide to go that route.

(1) use `/` notation:

```yaml
- property/get: System.Net.FtpWebRequest::KeepAlive
```

(2) use the `get_` and `set_` nomenclature seen in dnSpy:

```yaml
- property: System.Net.FtpWebRequest::get_KeepAlive
```

(3) don't facilitate specifying read/write:

```yaml
- property: System.Net.FtpWebRequest::KeepAlive
```

I lean towards `3` simply because I can't think of an example where it would be useful to specify read/write access.

**Comment by williballenthin** [comment]
do you think its important to differentiate properties from fields?

**Comment by mike-hunhoff** [comment]
No, I think it'd be fine to choose one feature term to represent both. If I were to choose one I'd prefer `field` or to reuse `offset` as you suggested.

**Comment by mr-tz** [comment]
I think (3) is sufficient for now. I'm not a fan of the other syntax proposals.

**Comment by mr-tz** [comment]
I've changed my mind and think differentiating between read and write would be useful.

## 1.
```yaml
- property/get: System.Net.FtpWebRequest::KeepAlive
- property/set: System.Net.FtpWebRequest::KeepAlive
```

## 2.
```yaml
- property/read: System.Net.FtpWebRequest::KeepAlive
- property/write: System.Net.FtpWebRequest::KeepAlive
```

## 3.
```yaml
- property/r: System.Net.FtpWebRequest::KeepAlive
- property/w: System.Net.FtpWebRequest::KeepAlive
```

I'm in favor of 2. since it's clear and appears most versatile (across languages/formats).

**Comment by mike-hunhoff** [comment]
closed by #1168
